### PR TITLE
fix: many rendering issues

### LINKED ISSUE #312: URL links do not get rendered inside bold text
**Describe the bug**
When a URL link is inside bold text, the URL does not get rendered.

**Setup**
Please complete the following information along with version numbers, if applicable.
 - OS: Ubuntu
 - Shell: zsh
 - Terminal Emulator: tilix
 - Terminal Multiplexer: tmux
 - Locale: en_AU

**To Reproduce**
1. Create a markdown `file.md` with the code in the source code section below
2. View the file `glow file.md`

**Source Code**
```
# File

**[URL](https://www.example.com)**
```

**Expected behavior**
Text in bold, followed by URL

**Screenshots**
glow-screenshot

### PR DESCRIPTION
**Pull request by caarlos0**
Fixing many bug related to rendering, most of them come from the fact we don't render inner elements properly. 
This fixes it.

closes https://github.com/charmbracelet/glamour/issues/312
closes https://github.com/charmbracelet/glamour/issues/290
closes https://github.com/charmbracelet/glamour/issues/106
closes https://github.com/charmbracelet/glamour/issues/274
closes https://github.com/charmbracelet/glamour/issues/311
closes https://github.com/charmbracelet/glamour/issues/239
closes https://github.com/charmbracelet/glamour/issues/262
closes https://github.com/charmbracelet/glamour/issues/254
closes https://github.com/charmbracelet/glamour/issues/315
closes https://github.com/charmbracelet/glamour/issues/316
refs https://github.com/charmbracelet/glamour/issues/149
closes https://github.com/charmbracelet/glamour/issues/117
closes https://github.com/charmbracelet/glamour/issues/60
closes https://github.com/charmbracelet/glamour/issues/313
closes https://github.com/charmbracelet/glamour/issues/237
refs https://github.com/charmbracelet/glamour/issues/107 (improves it a little bit, doesn't fix it though)
refs https://github.com/charmbracelet/glamour/issues/79 (improves it a little bit, doesn't fix it though)
closes https://github.com/charmbracelet/glamour/issues/68
refs https://github.com/charmbracelet/glamour/issues/172

### CODE REVIEW

**Review by aymanbagabas — APPROVED** [review]
Amazing!

**Review by maaslalani — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -4,23 +4,26 @@ go 1.21
 
 require (
 	github.com/alecthomas/chroma/v2 v2.14.0
+	github.com/charmbracelet/lipgloss v0.12.2-0.20240712161825-87dd58def709
```
Comment (aymanbagabas) [inline go.mod:7]: We probably should use a stable released version instead of pinning to a commit

**Review by aymanbagabas — APPROVED** [review]

**Comment by bashbunni** [comment]
I think this one is still broken
https://github.com/charmbracelet/glamour/issues/149
I tried reading the lipgloss README with glow and very long links were still broken
image

It also looks like there may be an issue with tables rendering properly
image

source markdown:
```

| Dec |  Hex |   Oct | Char | Description               |
| --: | ---: | ----: | :--- | ------------------------- |
| 001 | 0x00 | 0o000 | NUL  | null                      |
| 002 | 0x01 | 0o001 | SOH  | start of heading          |
| 003 | 0x02 | 0o002 | STX  | start of text             |
| 004 | 0x03 | 0o003 | ETX  | end of text               |
| 005 | 0x04 | 0o004 | EOT  | end of transmission       |
| 006 | 0x05 | 0o005 | ENQ  | enquiry                   |
```

We still need to fix these hard new lines as well... though could be done in another PR
image

**Comment by caarlos0** [comment]
this table breaks on glow, but not on glamour... weird, investigating...

**Pull request merged**